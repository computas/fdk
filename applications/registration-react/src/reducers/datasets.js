import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../constants/ActionTypes';

export default function datasets(
  state = { isFetchingDatasets: false, datasetItems: null },
  action
) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        isFetchingDatasets: true
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: action.response.data
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: null
      };
    default:
      return state;
  }
}
