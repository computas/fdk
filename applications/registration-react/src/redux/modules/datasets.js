import _ from 'lodash';
import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../../constants/ActionTypes';
import { fetchActions } from '../fetchActions';

function shouldFetch(metaState) {
  const threshold = 60 * 1000; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      (metaState.lastFetch || 0) < Date.now() - threshold)
  );
}

export function fetchDatasetsIfNeeded(catalogId) {
  return (dispatch, getState) =>
    shouldFetch(_.get(getState(), ['datasets', catalogId, 'meta'])) &&
    dispatch(
      fetchActions(`/catalogs/${catalogId}/datasets?size=100`, [
        { type: DATASETS_REQUEST, meta: { catalogId } },
        { type: DATASETS_SUCCESS, meta: { catalogId } },
        { type: DATASETS_FAILURE, meta: { catalogId } }
      ])
    );
}

const initialState = {};

export default function datasets(state = initialState, action) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        [action.meta.catalogId]: {
          meta: {
            isFetching: true,
            lastFetch: null
          }
        }
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        [action.meta.catalogId]: {
          items: _.get(action.payload, ['_embedded', 'datasets']),
          meta: {
            isFetching: false,
            lastFetch: Date.now()
          }
        }
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        [action.meta.catalogId]: {
          meta: {
            isFetching: false,
            lastFetch: null
          }
        }
      };
    default:
      return state;
  }
}

export const getDatasetItemsByCatalogId = (datasets, catalogId) =>
  _.get(datasets, [catalogId, 'items']);
