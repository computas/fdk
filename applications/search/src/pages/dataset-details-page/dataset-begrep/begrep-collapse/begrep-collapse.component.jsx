import React from 'react';
import PropTypes from 'prop-types';
import { Collapse } from 'reactstrap';

import './begrep-collapse.scss';

export class BegrepCollapse extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      detailed: false
    };
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({ detailed: !this.state.detailed });
  }

  render() {
    return (
      <div className="fdk-container-detail fdk-container-detail-begrep">
        <div
          className="fdk-ingress mb-0 fdk-container-begrep"
          role="button"
          tabIndex={0}
          onClick={this.toggle}
          onKeyPress={this.toggle}
        >
          <strong className="float-left">{this.props.prefLabel}:&nbsp;</strong>
          {this.props.note && (
            <i className="fa fa-chevron-down fdk-fa-right float-right" />
          )}
          <div>{this.props.definition}</div>
          <Collapse isOpen={this.state.detailed}>
            <div>{this.props.note}</div>
          </Collapse>
        </div>
      </div>
    );
  }
}

BegrepCollapse.defaultProps = {
  prefLabel: '',
  definition: '',
  note: ''
};

BegrepCollapse.propTypes = {
  prefLabel: PropTypes.string,
  definition: PropTypes.string,
  note: PropTypes.string
};
