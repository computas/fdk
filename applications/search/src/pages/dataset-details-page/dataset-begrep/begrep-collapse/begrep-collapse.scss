.fdk-container-begrep {
  padding: .5em 0;
}