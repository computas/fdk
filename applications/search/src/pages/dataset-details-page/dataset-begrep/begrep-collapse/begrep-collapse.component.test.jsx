import React from 'react';
import { shallow } from 'enzyme';
import { BegrepCollapse } from './begrep-collapse.component';
import datasets from '../../../../../test/fixtures/datasets';

test('should render BegrepCollapse correctly with no props', () => {
  const minWrapper = shallow(<BegrepCollapse />);
  expect(minWrapper).toMatchSnapshot();
});

test('should render BegrepCollapse correctly', () => {
  const { _source } = datasets[0];
  const defaultProps = {
    prefLabel: _source.subject[0].prefLabel.no,
    definition: _source.subject[0].definition.no,
    note: _source.subject[0].note.no
  };
  const wrapper = shallow(<BegrepCollapse {...defaultProps} />);
  wrapper.find('div.fdk-container-begrep').simulate('click');
  expect(wrapper.state('detailed')).toBe(true);
  expect(wrapper).toMatchSnapshot();
});
