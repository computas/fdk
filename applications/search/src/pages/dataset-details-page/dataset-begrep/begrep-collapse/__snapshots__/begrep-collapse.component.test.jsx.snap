// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BegrepCollapse correctly 1`] = `
<div
  className="fdk-container-detail fdk-container-detail-begrep"
>
  <div
    className="fdk-ingress mb-0 fdk-container-begrep"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <strong
      className="float-left"
    >
      enhet
      : 
    </strong>
    <i
      className="fa fa-chevron-down fdk-fa-right float-right"
    />
    <div>
      alt som er registrert med et organisasjonsnummer 
    </div>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <div>
        Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.
      </div>
    </Collapse>
  </div>
</div>
`;

exports[`should render BegrepCollapse correctly with no props 1`] = `
<div
  className="fdk-container-detail fdk-container-detail-begrep"
>
  <div
    className="fdk-ingress mb-0 fdk-container-begrep"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <strong
      className="float-left"
    >
      : 
    </strong>
    <div />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <div />
    </Collapse>
  </div>
</div>
`;
